initfiles/sbin/hpcc_setenv.in

```diff
@@ -23,12 +23,15 @@
 ###<REPLACE>###
 
 function kill_process () {
-    SENTINEL=$1
-    PID=$2
-    TIMEOUT=$3
-    if [ -e $SENTINEL ]; then
-        rm -f $SENTINEL
+    if [ $# -eq 3 ]; then
+        SENTINEL=$1
+        shift
+        if [ -e $SENTINEL ]; then
+            rm -f $SENTINEL
+        fi
     fi
+    PID=$1
+    TIMEOUT=$2
```

Comment:
If 3 parameters then 1st is sentinel, 2nd is pid ...
if <3 , 1st is pid.. ?

I think that's confusing.

Why not optional 3rd parameter = sentinel ?